File: packages/disband-bot-python/disband_bot/message.py
"""Message model for Disband bots."""


class Message:
    """A message received from, or sent to, a Disband channel."""

    def __init__(self, data, client):
        self.client = client
        self.id = data.get("id")
        self.channel_id = data.get("channel_id")
        self.server_id = data.get("server_id")
        self.author = data.get("author") or None
        self.content = data.get("content") or ""
        self.reply_to_id = data.get("reply_to_id")
        self.mentions = data.get("mentions") or []
        attachment_url = data.get("attachment_url")
        self.attachment = (
            {"url": attachment_url, "type": data.get("attachment_type")}
            if attachment_url
            else None
        )
        self.created_at = data.get("created_at")
        self.edited_at = data.get("edited_at")
        self.display_id = data.get("display_id")

    @property
    def author_is_bot(self):
        author = self.author or {}
        return bool(author.get("is_bot"))

    def reply(self, content):
        """Replies to this message in the same channel."""
        return self.client.send_message(self.channel_id, content, reply_to_id=self.id)

    def __repr__(self):
        return "<Message id=%s channel=%s content=%r>" % (
            self.id,
            self.channel_id,
            self.content[:40],
        )

File: packages/bot/src/message.js
/** A message as returned by the Disband API. */
export class Message {
  constructor(data, client) {
    this.client = client;
    this.id = data.id;
    this.channelId = data.channel_id;
    this.serverId = data.server_id ?? null;
    this.author = data.author ?? null;
    this.content = data.content ?? "";
    this.replyToId = data.reply_to_id ?? null;
    this.mentions = data.mentions ?? [];
    this.attachment = data.attachment_url
      ? { url: data.attachment_url, type: data.attachment_type ?? null }
      : null;
    this.createdAt = data.created_at ?? null;
    this.editedAt = data.edited_at ?? null;
    this.displayId = data.display_id ?? null;
  }

  /** True when the message was sent by a bot account. */
  get authorIsBot() {
    return Boolean(this.author?.is_bot);
  }

  /** Replies to this message in the same channel. */
  async reply(content, options = {}) {
    return this.client.sendMessage(this.channelId, content, {
      ...options,
      replyToId: options.replyToId ?? this.id,
    });
  }
}
